src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -281,8 +281,17 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
     return;
   }
 
-  auto lease_client = GetOrConnectLeaseClient(raylet_address);
   TaskSpecification &resource_spec = task_queue.front();
+  if (raylet_address == nullptr) {
```

Comment:
The clunkiness around the `raylet_address` API/semantics will be gone once [locality-aware spillback](https://github.com/ray-project/ray/issues/12815) is implemented.